content/articles/servlet-request-and-servlet-collaboration/SevletRequest and ServletCallaboration.md

```diff
@@ -0,0 +1,277 @@
+The service container creates ServletRequest and ServletResponse objects when a client sends a request to a web server, which passe as an argument in the service method.
+ The request object allows access to request information such as the header and the requested data body information.
+
+
+### Methods of ServletRequest interface
+The ServletRequest interface provides us with many methods to work with client request information. Some explained as follows.
+
+- `String getParameter(String variableName)`: This method returns a value of parameter requested by a client of a return type `String`. 
+
+- `String[] getParameterValues(type parameterName)`: Majorly used to obtained all values of a given multi-valued parameter name. In this case, useful while working with the multi-select list box where a single parameter may contain multiple values. 
+
+- `public int getContentLength()`: It  returns the actual size of requested data of type `int`.
+
+- `Public String getContentType()`:  Returns the data of a request Internet Media Type, or null, if unknown. 
+
+- `Enumeration getParameterNames()`: It’s used to return an enumeration of all target parameter name.
+
+- `String getCharacterEncoding()`: This method returns the name if the encoded character is specified. If not specified, it returns null in the body of the request.
+
+- `public int getServerPort()`: This method returns the port number that received this request.
+
+- ``public abstract String getServerName()``:  It returns the server name which receives the request.
+
+- `ServletInputStream getInputStream(): throws IOException`: Returns the binary data input stream in the query body. 
+
+ - `String getCharacterQueryString()`: Returns the url path of the request from the protocol name of a type `String.`
+
+- `getInputStream()`: This method returns information stream from the standard output stream of a type `ServletInputStream`
+
+- ``String getLocalAddr()``: It returns the IP address of the interface from which the request has taken.
+
+- `getAttribute(String name)`:  It returns the attribute of the requested object set by its name, and the return type is `Object.`
+
+- `Enumeration getAttributeName()`: It returns the Attribute name of the type Enumeration available in the current request.
+
+- `ServletContext getServletContext()`: It returns the content of the current Servlet request and the return type is `ServletContext`.
+
+- `removeAttribute(String name)`: Used when removing the attribute from the current request of a type `void.`
+
+### Example of ServletRequest interface
+In this example, we are going to create a user login project using servlet. We used the getParameter method () to return the value of the specified request parameter name in this context.
+
+**Steps to create this project.**
+

```

Comment:
for your steps I suggest this format :

#### Step 1:
Open the IDE...
#### Step 2:
blablabla ......